src/watch/fileWatchers.ts

```diff
@@ -72,15 +67,24 @@ export default class FileWatcher {
 		const handleWatchEvent = (event: string) => {
 			if (event === 'rename' || event === 'unlink') {
 				this.fsWatcher.close();
-				this.trigger();
-				dispose();
+				this.trigger(id);
+				group.delete(id);
 			} else {
-				// this is necessary because we get duplicate events...
-				const contents = fs.readFileSync(id, 'utf-8');
-				if (contents !== data) {
-					data = contents;
-					this.trigger();
+				let stats: fs.Stats;
+				try {
+					stats = fs.statSync(id);
+				} catch (err) {
+					if (err.code === 'ENOENT') {
+						if (modifiedTime !== -1) {
+							modifiedTime = -1;
+							this.trigger(id);
+						}
+						return;
+					}
+					throw err;
 				}
+				// debounce
+				if (+stats.mtime - modifiedTime > 50) this.trigger(id);
```

Comment:
The goal of a debounce is to remove noisy duplicate events from an event stream, that hold no significant meaning.

An automated process that changes a file for good reason is not an example of a debounce because typically that processing time should be longer than the debounce threshold.

Since the debounce needs to take into account the file modified state, it seems sensible to keep it with the individual file watcher instance as that class is associated with the file's state.

Reply:
I've lowered the debounce thresold and renamed the private methods.